clang/lib/Sema/SemaOverload.cpp

```diff
@@ -10960,6 +10960,10 @@ OverloadCandidateSet::BestViableFunction(Sema &S, SourceLocation Loc,
     S.diagnoseEquivalentInternalLinkageDeclarations(Loc, Best->Function,
                                                     EquivalentCands);
 
+  // [over.match.best]/4 is checked for in Sema::ConvertArgumentsForCall,
+  // because not every function call goes through our overload resolution
+  // machinery, even if the Standard says it supposed to.
+
```

Comment:
do we have test for something like that

```
namespace A {
    extern "C" void f(long = 0); // viable candidate
}
namespace B {
    extern "C" void f(long = 0); // viable candidate
}

using A::f;
using B::f;

void f(int) {} // best candidate

void g() {
    f(0); // should not diagnose
}
```

Reply:
ideally we would return `OR_Ambiguous` here, and then special case the diagnostic for that when all candidates are redeclarations

Reply:
We have `f(0)` test in DR tests, but I don't understand how `void f(int) {} // best candidate` would bring additional coverage. I also don't understand why we'd return `OR_Ambiguous`.

I think it's worth pointing out that when there is a redeclaration chain, only one declaration out of it is considered a candidate function for overload resolution. At least as far as I saw in the debugger while working on this.